File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0;
    let minSoFar = prices[0]

    for (let i=1; i<prices.length; i++){
        minSoFar = Math.min(minSoFar,prices[i])
        maxProfit = Math.max(maxProfit,prices[i]-minSoFar)
    }
    return maxProfit
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice=prices[0]
        income=0
        for i in range(1,len(prices)):
            minPrice=min(minPrice,prices[i])
            income=max(income,prices[i] - minPrice)
        return income
        